core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -113,6 +115,143 @@ public static Object getMeasureValueBasedOnDataType(String msrValue, DataType da
     }
   }
 
+  public static Object getMeasureObjectFromDataType(byte[] data, DataType dataType) {
+    ByteBuffer bb = ByteBuffer.wrap(data);
```

Comment:
This is unnecessary object for `decimal` so keep it in respective case statement